tests/apps/test_embed.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Unit tests for strawberryfields.apps.train.embed
+"""
+# pylint: disable=no-self-use,unused-argument
+import numpy as np
+import pytest
+from strawberryfields.apps.train import embed
+
+pytestmark = pytest.mark.apps
+
+feats = [
+    [[0.1, 0.2], [0.2, 0.1]],
+    [[0.1, 0.2, 0.3], [0.3, 0.1, 0.2], [0.2, 0.3, 0.1]],
+    [[0.1, 0.2, 0.3, 0.4], [0.4, 0.1, 0.2, 0.3], [0.3, 0.4, 0.1, 0.2], [0.2, 0.3, 0.4, 0.1]],
+]
+feats = np.array([np.array(f) for f in feats])
+
+ps = [[1.0, 2.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0, 4.0]]
+ps = np.array([np.array(p) for p in ps])
+
+weights_f = np.array(
+    [
+        np.exp(-np.array([0.5, 0.4])),
+        np.exp(-np.array([1.4, 1.1, 1.1])),
+        np.exp(-np.array([3.0, 2.4, 2.2, 2.4])),
+    ]
+)
+weights = np.array(
+    [
+        np.exp(-np.array([1.0, 2.0])),
+        np.exp(-np.array([1.0, 2.0, 3.0])),
+        np.exp(-np.array([1.0, 2.0, 3.0, 4.0])),
+    ]
+)
+
+jacobian_f = np.array([np.zeros((d, d)) for d in range(2, 5)])
+jacobian = np.array([np.zeros((d, d)) for d in range(2, 5)])
+
+for i in range(3):
+    jacobian[i] = -np.diag(weights[i])
+    for j in range(i + 2):
+        for m in range(i + 2):
+            jacobian_f[i][j, m] = -feats[i][j, m] * weights_f[i][j]
+
+
+@pytest.mark.parametrize("dim", range(2, 5))
+class TestExpFeatures:
+    """Tests for the class ``strawberryfields.apps.train.embed.ExpFeatures``"""
+
+    def test_invalid_dim(self, dim):
+        """Tests that a warning is issued if the feature vectors do not have the correct
+        dimension"""
+        with pytest.raises(ValueError, match="Dimension of parameter vector"):
+            features = np.ones((dim, dim + 1))
+            expf = embed.ExpFeatures(features)
+            expf(np.zeros(dim))
+
+    @pytest.mark.parametrize("m", range(2, 5))
+    def test_exp_features_integration(self, dim, m):
+        """Test that ``strawberryfields.apps.train.embed.ExpFeatures`` outputs weights and jacobian
+        of the correct shape"""
```

Comment:
```suggestion
    def test_exp_features_shape(self, dim, m):
        """Tests that the weights and jacobian have the correct shape for a range of different 
        sizes for the input features matrix"""
```
^ Just to make it fit in a bit more with the style of the other test methods